Sources/NIO/System.swift

```diff
@@ -625,8 +627,26 @@ internal enum KQueue {
     @inline(never)
     @discardableResult
     public static func kevent(kq: CInt, changelist: UnsafePointer<kevent>?, nchanges: CInt, eventlist: UnsafeMutablePointer<kevent>?, nevents: CInt, timeout: UnsafePointer<Darwin.timespec>?) throws -> CInt {
+        var timeoutClamped = timeout
+        // Note that a nil timeout waits indefinitely
+        if let ts = timeout {
+            // Assert to check that timeout conforms to the timespec definition in the C11 standard (ISO/IEC 9899:2011)
+            assert(ts.pointee.tv_nsec >= 0 || ts.pointee.tv_nsec < 1_000_000_000, "kevent timeout value is malformed (0 <= tv_nsec < 1_000_000_000)")
+            // Validate that the timeout parameter passed to kevent does not exceed the limits of the Darwin kernel
+            // interval timer (note that - whilst unlikely - these values could change)
+            if ts.pointee.tv_sec < 0  || ts.pointee.tv_sec > sysIntervalTimerMaxSec {
+                let tsClamped = UnsafeMutablePointer(mutating: ts)
+
+                // Clamp the timeout tv_sec value to the maximum accepted by the kernel
+                // (note this schedules the event many years in the future)
+                tsClamped.pointee.tv_sec = sysIntervalTimerMaxSec
+
+                timeoutClamped = UnsafePointer(tsClamped)
```

Comment:
We can't do this. This is the Swift equivalent of casting away `const` in C, with the same affects. The Swift compiler can and will assume that this function will not mutate through this pointer, and doing so may invoke undefined behaviour.

In this case, the issue will probably go away if we hoist this logic higher up in the code as we'll probably have the `timespec` struct anyway, and so can just mutate it.